### PR TITLE
Concourse pipeline

### PR DESCRIPTION
**Pull request by ccemeraldeyes**

### CODE REVIEW

**Comment by ccemeraldeyes** [comment]
@evandbrown please review. I can't seem to assign to you.

File: ci/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    type: docker-image
+    source:
+      repository: frodenas/gcs-resource
+
+resources:
+  - name: gcp-service-broker
+    type: git
+    source:
+      uri: https://github.com/GoogleCloudPlatform/gcp-service-broker.git
+      branch: concourse
```
Comment (evandbrown) [inline ci/pipeline.yml]: Is the `concourse` branch like a `develop` branch? We use `develop` and `master` in other projects.

File: ci/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+  - name: version
+    type: semver
+    source:
+      bucket: gcp-service-broker-semver
+      key: semver
+      endpoint: storage.googleapis.com
+      access_key_id: {{storage_key_id}}
+      secret_access_key: {{storage_secret_key}}
+      initial_version: 3.1.3
```
Comment (evandbrown) [inline ci/pipeline.yml]: Is this really the version we're starting from?

File: ci/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+resources:
+  - name: gcp-service-broker
+    type: git
+    source:
+      uri: https://github.com/GoogleCloudPlatform/gcp-service-broker.git
+      branch: concourse
+      ignore_paths:
+        - ci/*
+
+  - name: gcp-service-broker-ci
```
Comment (evandbrown) [inline ci/pipeline.yml]: Why 2 resources for the same git repo/branch? I see one ignores the `ci/` dir and the other doesn't. Possible to use just 1 and avoid cloning twice?

File: ci/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      endpoint: storage.googleapis.com
+      access_key_id: {{storage_key_id}}
+      secret_access_key: {{storage_secret_key}}
+      initial_version: 3.1.3
+
+jobs:
+  - name: test-unit
+    plan:
+      - aggregate:
+        - get: gcp-service-broker
```
Comment (evandbrown) [inline ci/pipeline.yml]: This is aggregating the two resources that are the same git repo. Is that necessary?

File: ci/pipeline.yml (hunk 1 above)
Comment (ccemeraldeyes) [inline ci/pipeline.yml]: yes, once this is merged I'll change the branch to master.

File: ci/pipeline.yml (hunk 3 above)
Comment (ccemeraldeyes) [inline ci/pipeline.yml]: it's essentially a way to separate out the ci config as a resource. The fact that it's in the same repo is just how we've been setting things up, but really the code and the ci config should be considered different resources (which also allows for out-of-sync updating).

File: ci/pipeline.yml (hunk 2 above)
Comment (ccemeraldeyes) [inline ci/pipeline.yml]: yup! One later than the version that's public now.

File: ci/pipeline.yml (hunk 4 above)
Comment (ccemeraldeyes) [inline ci/pipeline.yml]: yes, so the CI one is independent.

**Pull request merged**